inbm/cloudadapter-agent/cloudadapter/cloud/client/inbs_cloud_client.py

```diff
@@ -0,0 +1,181 @@
+"""
+Modification of Cloud Client class that works for INBS.
+INBS is different because it is using gRPC instead of MQTT.
+"""
+
+import queue
+import threading
+import time
+from typing import Callable, Optional, Any
+from datetime import datetime
+from ..adapters.proto import inbs_sb_pb2_grpc, inbs_sb_pb2
+import logging
+
+import grpc
+from .cloud_client import CloudClient
+
+logger = logging.getLogger(__name__)
+
+
+class InbsCloudClient(CloudClient):
+
+    def __init__(self,
+                 hostname: str,
+                 port: str,
+                 node_id: str,
+                 token: str) -> None:
+        """Constructor for InbsCloudClient
+        """
+
+        self._grpc_hostname = hostname
+        self._grpc_port = port
+        self._client_id = node_id
+        self._metadata = [
+            ("node-id", node_id),
+            ("token", token)
+        ]
+
+        self._stop_event = threading.Event()
+
+    def get_client_id(self) -> Optional[str]:
+        """A readonly property
+
+        @return: Client ID
+        """
+
+        return self._client_id
+
+    def publish_telemetry(self, key: str, value: str, time: datetime) -> None:
+        """Publishes telemetry to the cloud
+
+        @param key: telemetry's key to publish
+        @param value: data to publish to the telemetry
+        @param time: timestamp for this telemetry publish
+        @exception PublishError: If publish fails
+        """
+
+        raise NotImplementedError("publish_telemetry not yet implemented")
```

Comment:
<h3>$\color{#FFA500}Improvements: Code Refactoring$</h3>

Implement the `publish_telemetry`, `publish_event`, and `publish_attribute` methods or provide a clear documentation on why they are not implemented if they are not applicable for INBS.

Reply:
good idea; let's explain that these are coming later in our roadmap